### ISSUE TITLE
Update source code

### ISSUE THREAD

**Issue by onli** [pipes-digital/pipes#79]
Pipes CE and pipes.digital diverged. This is because I made an error in the beginning: When I created Pipes CE, I did not want to have the pricing page and the full git history of the original repo available, so Pipes CE was basically a reduced fork. But that lead to fixes regularly going only into the separate git project for pipes.digital - which means the code here is in worse shape. Additionally the project looks stale, which at the very least hurts my pride.

Solution: Do it right this time. Take the source code of pipes.digital, modularize it so that the parts a selfhosted pipes does not need are simply not loaded by default (but still available). Blocks that rely on additional hosted infrastructure should be toggled by environment variables pointing to the URL of the needed service (like the reddit integration).

**Comment by dentarg** [comment]
Yeah, what is the `thread` gem? It is not on RubyGems (anymore?): https://rubygems.org/gems/thread

https://github.com/pipes-digital/pipes/blob/38e7dbd4a3c2452b47611f5c78d652318ee88c15/Gemfile.lock#L130

**Comment by onli** [comment]
@dentarg I'm sorry I never responded here. Not sure why the link you showed does not work, but I see https://rubygems.org/gems/thread/versions/0.2.2 - and that seems to be installable.
